sensorprocess/sensorprocess.go

```diff
@@ -47,24 +48,27 @@ func Start(
 func addSensorReading(
 	ctx context.Context,
 	config Config,
+	isOnline bool,
 ) bool {
 	tsr, err := config.Lidar.TimedSensorReading(ctx)
 	if err != nil {
 		config.Logger.Warn(err)
-		return strings.Contains(err.Error(), replaypcd.ErrEndOfDataset.Error())
+		isEndOfDataset := strings.Contains(err.Error(), replaypcd.ErrEndOfDataset.Error())
+		jobDone := isEndOfDataset && !isOnline
+		return jobDone
```

Comment:
This change makes it so we do not end the sensor process if we are in offline mode - should we sleep here?

Reply:
Oops, I missed this - Can we check data rate msec from `config` here and end the sensor process if we get back ErrEndOfDataset and the data rate msec is 0?

Reply:
`isOnline` comes from data_rate_msec here. Do you want the whole config to be passed in instead?

Reply:
nvm got it!